snapcraft/models/project.py

```diff
@@ -2064,7 +2064,7 @@ def get_extra_build_snaps(self) -> list[str]:
         extra_build_snaps = list(content_snaps - part_build_snaps)
 
         # Always add the base as an extra build snap
-        if self.base is not None:
+        if self.base is not None and self.build_base == self.base:
```

Comment:
What would you expect to be installed in the build environment for this snapcraft.yaml?

```yaml
build-base: core22
base: core22-desktop
```

Reply:
Is `coreXX-desktop` still a thing? I thought we abandoned that.

Reply:
> Is coreXX-desktop still a thing? I thought we abandoned that.

I'm not sure, we were requested to support desktop bases last year (https://github.com/canonical/snapcraft/issues/4818).  Do you know when it was abandoned or if that's in documented somewhere? 

> What about `if self.base is not None and self.build_base != "devel"`

Can you point me to the snapcraft.yaml you're using?

I'm wondering if this will be the correct behavior in the period of time leading up to the stable release of core26.  When core26 is available in a non-stable channel and snap developers are doing early testing of their snap on core26 with:
```yaml
base: core24
build-base: devel
```
then core26 won't get installed in their build environment.

Reply:
> > Is coreXX-desktop still a thing? I thought we abandoned that.
> 
> I'm not sure, we were requested to support desktop bases last year (#4818). Do you know when it was abandoned or if that's in documented somewhere?

Maybe we should ask @jhenstridge

Reply:
@mr-cal hi there, I can confirm that having bases specific for desktop was nack'ed and the plan for core desktop is to use coreXX in the future. So no more coreXX-desktop bases will be `produced.